src/activities/DeleteCartegraphRecord.ts

```diff
@@ -62,7 +63,7 @@ export default class DeleteCartegraphRecord implements IActivityHandler {
         const url = new URL(
             `${service.url}/api/v1/classes/${encodeURIComponent(className)}`,
         );
-        if (id) {
+        if (typeof id === "number") {
             url.pathname += `/${id}`;
         }
         filter && url.searchParams.append("filter", filter);
```

Comment:
Do we want to urlEncode parameters or does searchParams take care of that for us?